planner/core/rule_partition_processor.go

```diff
@@ -142,3 +150,13 @@ func (s *partitionProcessor) canBePrune(ctx sessionctx.Context, col *expression.
 	}
 	return len(r) == 0, nil
 }
+
+// findByName checks whether object name exists in list.
+func findByName(list []model.CIStr, name string) bool {
```

Comment:
1. Bind it to `partitionProcessor`?
2. s/list/partitionNames/ ?
3. s/name/partitionDef/ ?

Reply:
Why bind to `partitionProcessor`?

Reply:
If a function is not intend to be called by multiple objects from different structs, I think it's a good practice to bind it to the struct which calls it.

Reply:
done.